release.md

```diff
@@ -0,0 +1,73 @@
+# Release procedure
+
+## Branching
+
+1. Checkout `master` branch.
+
+    `git checkout master`
+1. Make sure you have the latest upstream changes.
+
+    `git pull upstream master`
+
+1. Create a new release branch.
+
+    `git checkout -b v1.x`
+1. Create branch on the upstream
+
+    `git push upstream v1.x`
+
+1. Set branch protection rules on `v1.x` branch in GitHub repository settings.
+
+1. Change image tag in Makefile.
+
+    `sed -i 's/IMAGE_TAG ?= latest/IMAGE_TAG ?= 1.x/g' Makefile`
+1. Update manifests, examples and chart defaults.
+
+    `make update`
+1. Save changes and create a PR with `v1.x` as base branch.
+
+    `git commit -a -m "Prepare v1.x-rc.0`
+1. Once PR is merged, tag release candidate using instructions below.
+
+1. Checkout `master` branch.
+
+    `git checkout master`
+
+1. Update release schedule in `docs/source/release.md`.
+
+1. Once PR with release schedule and enabled docs is merged to master, tag merge commit as `alpha.0` of next expected release.
+
+1. Enable building docs from `v1.x` branch by adding entry to `docs/source/conf.py`.
+
+    `BRANCHES = ['master', 'v0.3', 'v1.0', 'v1.1', 'v1.x']`
+1. Submit PR with additional docs branch with `master` as target branch.
+
+## Tagging
+
+1. Tag commit using annotated tag
+
+    `git tag v1.x.0-rc.0 <commit-sha> -m "Release v1.x-rc.0`
+1. Prepare release notes using new tag and latest related version tag.
+
+    `make build && ./gen-release-notes --ref-start v1.x.0-alpha.0 --ref-end v1.x.0-rc.0 --release-name v1.x.0-rc.0`
+
+1. Publish release notes on users mailing list and Slack channel.
+
+## Publishing a final release
+
+1. Tag a final release tag using above instructions.
```

Comment:
Would and explicit ref be safer so we always tag the same commit?
`git tag vX.X.0 vX.X.0-rc.X` -a -m ''vX.X.0"